design/notify.go

```diff
@@ -29,6 +29,9 @@ var notificationAttributes = a.Type("NotificationAttributes", func() {
 	a.Attribute("custom", a.HashOf(d.String, d.Any), "custom information to be passed as a json", func() {
 		a.Example(map[string]interface{}{"verification_url": "https://auth.openshift.io/api/user/verifyEmail?code=bb9148f7"})
 	})
+	a.Attribute("revisionID", d.UUID, "ID of revision that triggered this notification", func() {
+		a.Example("1646e178-b90e-48b4-85e4-204d1c4c416f")
+	})
 	a.Required("type", "id")
```

Comment:
With comments update also it will be used!

Reply:
@surajssd my concern is_not that `revisionID` is useful for how many notification type.  My concern is with `revisionID` design.

How about this design - rather `revisionID` string we can introduce `identity` map.

So JSON request paylod would be:
```
{
  "data": {
    "attributes": {
      "id": "guid_id",
      "identity": {
        "revisionID": "revision_id"
      },
      "type": "workitem.update"
    },
    "ID": "2a35344c-43de-4767-bf3b-b951d22bf7df",
    "type": "notifications"
  }
}
```

And `ReceiverResolver` definition would be:
```
type ReceiverResolver func(context.Context, string, interface{}) (users []Receiver, templateValues map[string]interface{}, err error)
```

And WorkItemResolver will extract `revisionID` from `identity`.  And rest ReceiverResolver will just ignore it.

This design will allow to accommodate some other param (similar to `revisionID`) come in future.  It will just go inside `identity` map.

Reply:
why are we protecting the param addition here and trying to also off load the heavy lifting to some generic types?

This will just make the API brittle IMHO. We should either namespace further this API or add concrete parameters.

Also the whole reason we are using goa is so that we don't have to write all this validation logic on the data that is being sent, etc. and we are getting around that to do what goa does one layer above, then why even use goa?

Reply:
>or... the ID is the rev id. Don't really need both..

We need both at the moment, because wit has no way of telling you anything with just the revision id/event id. What we agreed upon is that we use the same endpoint to list all the events and then filter here. So that's what the plan if for now.

Reply:
> if no revision it could be validated to be a failed notification or default to latest

what do you mean by the `default to latest` ?

Reply:
@surajssd what I understood is:
add rev_id under "custom" and in validator check if rev_id exists?  here two possibilities, first if rev_id not exists then fail the call.  second, if rev_id not exists then find out latest rev_id and use it.